pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -301,6 +262,25 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(cb *ClusterBuilder,
 	return resources, cacheStats{hits: hit, miss: miss}
 }
 
+func skipForDelta(key model.XdsCacheEntry, updated map[model.ConfigKey]struct{}) bool {
```

Comment:
This is the main difference from #37465 

But the updated configs may include deleted configs, not sure how you will handle this

Reply:
Yeah that will be the tricky part. Maybe if this functionsl returns `true` but the cluster generator doesn't build a cluster, that is a "delete". Would have to test it